airbyte-webapp/src/hooks/useTrackAction.ts

```diff
@@ -2,25 +2,68 @@ import { useCallback } from "react";
 
 import { useAnalyticsService } from "./services/Analytics/useAnalyticsService";
 
-export const enum TrackActionType {
+export const enum LegacyTrackActionType {
   NEW_SOURCE = "New Source",
   NEW_DESTINATION = "New Destination",
+  NEW_CONNECTION = "New Connection",
+  SOURCE = "Source",
 }
 
-interface TrackActionProperties {
+export const enum TrackActionActions {
+  CREATE = "Create",
+  TEST = "Test",
+  SELECT = "Select",
+  SUCCESS = "TestSuccess",
+  FAILURE = "TestFailure",
+  FREQUENCY = "FrequencySet",
+  SYNC = "FullRefreshSync",
+  SCHEMA = "EditSchema",
+  DISABLE = "Disable",
+  REENABLE = "Reenable",
+  DELETE = "Delete",
+}
+
+export const enum TrackActionNamespace {
+  SOURCE = "Source",
+  DESTINATION = "Destination",
+  CONNECTION = "Connection",
+}
+
+interface TrackConnectorActionProperties {
   connector_source?: string;
   connector_source_definition_id?: string;
   connector_destination?: string;
   connector_destination_definition_id?: string;
 }
 
-export const useTrackAction = (type: TrackActionType) => {
+interface TrackConnectionActionProperties {
+  frequency: string;
+  connector_source_definition: string;
+  connector_source_definition_id: string;
+  connector_destination_definition: string;
+  connector_destination_definition_id: string;
+  available_streams: number;
+  enabled_streams: number;
+}
+
+export const useTrackAction = (namespace: TrackActionNamespace, legacyType?: LegacyTrackActionType) => {
```

Comment:
Would it be possible to create a unit test for this hook?

Reply:
I'd looked into implementing it that way, but not all calls followed the `New {namespace} - action`.  Most of the Source calls were `{namespace} - action`.  Given, we know what those exceptions are, so we could store the namespace/actionType combinations that are exceptions in a list in the action and check against that? Thoughts?  Seemed overly complex to me but I'm open to it.

Reply:
> Would it be possible to create a unit test for this hook?
I think TypeScript has us pretty safeguarded against writing bad calls here, but testing it sounds right.  Anything that stands out as a good idea to test for explicitly?

Reply:
To ensure that the analyticsService is called with the correct parameters based on the different inputs (with and without legacy type, etc) - this is mainly to ensure that the hook does follow the tracking spec.